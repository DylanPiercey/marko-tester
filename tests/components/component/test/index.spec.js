const { render, fixtures } = require('../../../../src/index')('../index.marko', { withoutFixtures: true });

fixtures();

describe('When component is rendered with data', () => {
  let component;
  const cachedFixture = JSON.stringify(fixtures.default);
  const mockUtil = 'mockUtil';

  beforeEach(() => {
    jest.mock('../../non-marko', () => mockUtil);

    component = render(fixtures.default);
  });

  afterEach(() => {
    component.destroy();
  });

  it('should not mutate the fixture', () => {
    expect(JSON.stringify(fixtures.default)).toBe(cachedFixture);
  });

  it('should mock relative module', () => {
    expect(component.util).toBe(mockUtil);
  });

  it('should not render other component', () => {
    expect(component.getEl('tbody').tagName).toBe('TBODY-COMPONENT');
  });

  describe('When user clicks the button', () => {
    beforeEach((done) => {
      component.getEl('button').click();
      component.once('update', done);
    });

    it('should set change the state and rerender', () => {
      expect(component.state.hidden).toBe(true);
      expect(component.getEl('tbody')).toBe(undefined);
    });
  });
});

describe('When component is rendered without data', () => {
  let component;
  const mockUtil = 'mockUtil2';

  beforeEach(() => {
    jest.mock('../../non-marko', () => mockUtil);

    component = render(fixtures.hidden);
  });

  afterEach(() => {
    component.destroy();
  });

  it('should mock relative module', () => {
    expect(component.util).toBe(mockUtil);
  });
});

describe('When component is rendered without fixture', () => {
  let component;

  beforeEach(() => {
    component = render();
  });

  afterEach(() => {
    component.destroy();
  });

  it('should not error', () => {
    expect(true);
  });
});
