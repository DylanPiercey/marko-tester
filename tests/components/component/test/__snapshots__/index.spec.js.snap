// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component with default fixture 1`] = `
Array [
  ,
  <div
    class="mock-component"
  >
    <h1>
      hello
    </h1>
    <button
      type="button"
    >
      Hello
    </button>
    <tbody-component>
      <at_body>
        body is here
      </at_body>
    </tbody-component>
  </div>,
  ,
]
`;

exports[`should render component with hidden fixture 1`] = `
Array [
  ,
  <div
    class="mock-component"
  >
    <h1>
      
    </h1>
    <button
      type="button"
    >
      Hello
    </button>
  </div>,
  ,
]
`;

exports[`should render component with undefined fixture 1`] = `
Array [
  ,
  <div
    class="mock-component"
  >
    <h1>
      
    </h1>
    <button
      type="button"
    >
      Hello
    </button>
    <tbody-component>
      <at_body>
        body is here
      </at_body>
    </tbody-component>
  </div>,
  ,
]
`;
